adijif/clocks/ad9545.py

```diff
@@ -0,0 +1,330 @@
+                    self.model.Minimize(self.model.abs3(self.config["PLL" + str(i) + "_rate"] - average_PLL_rate))
+                else:
+                    raise Exception("Unknown solver {}".format(self.solver))
+
+        """ Instruct solver to maximize PLL input rates """
+        for i in range(0, 4):
+            if self.input_refs[i] != 0:
+                if self.solver == "CPLEX":
+                    self.model.maximize(self.config["PLL_in_rate_" + str(i)])
+                elif self.solver == "gekko":
+                    self.model.Maximize(self.config["PLL_in_rate_" + str(i)])
+                else:
+                    raise Exception("Unknown solver {}".format(self.solver))
+
+    def _setup(self, input_refs: List[int], out_freqs: List[int]) -> None:
+        # Setup clock chip internal constraints
+
+        self._setup_solver_constraints(input_refs, out_freqs)
+
+    def set_requested_clocks( self, input_refs: List, out_freqs: List, clk_names: List[str] ) -> None:
+        """Define necessary clocks to be generated in model.
+
+        Args:
+            out_freqs (List): list of required clocks to be output(input_refs[i] / self.config["r" + str(i)])
+            clk_names (List[str]):  list of strings of clock names
+
+        Raises:
+            Exception: If len(out_freqs) != len(clk_names)
+        """
+
+        if len(clk_names) != len(out_freqs):
+            raise Exception("clk_names is not the same size as out_freqs")
+        self._clk_names = clk_names
+
+        self._setup(input_refs, out_freqs)
+
+        """ Add output dividers as variables to the model """
+        for i in range(0,10):
+            if out_freqs[i] != 0:
+                if self.solver == "gekko":
+                    self.config["q" + str(i)] = self.model.Var(integer=True, lb=self.Q_min, ub=self.Q_max)
+                    self.config["out_rate_" + str(i)] = self.model.Const(int(out_freqs[i]))
+                elif self.solver == "CPLEX":
+                    self.config["q" + str(i)] = exp.integer_var(domain=(self.Q_min, self.Q_max))
+                    self.config["out_rate_" + str(i)] = exp.CpoValue(value=int(out_freqs[i]), type=ctg.Type_Int)
+                else:
+                    raise Exception("Unknown solver {}".format(self.solver))
+
+        for i in range(0, 10):
+            if out_freqs[i] != 0:
+                if i > 5:
+                    pll_number = 1
+                else:
+                    pll_number = 0
+
+                self._add_equation(
+                    [
+                        self.config["PLL" + str(pll_number) + "_rate"] / self.config["q" + str(i)] == self.config["out_rate_" + str(i)]
+                    ]
+                )
+
+    def _solve_gekko(self) -> bool:
+        """Local solve method for clock model.
+
+        Call model solver with correct arguments.
+
+        Returns:
+            bool: Always False
+        """
+        self.model.options.SOLVER = 1  # APOPT solver
```

Comment:
Do that default settings from clock.py not work for gekko?

Reply:
With the default settings from clock.py , for some inputs i get solution not found.

Reply:
If you update clocks.py to these settings do other tests fail?

Reply:
So i added this in clock.py: 
        self.model.options.SOLVER = 1  # APOPT solver
        self.model.solver_options = [
            "minlp_maximum_iterations 300000000",
            "minlp_max_iter_with_int_sol 6000",
            "minlp_as_nlp 0",
            "minlp_branch_method 1",
            "minlp_integer_tol 0.0001"
        ]
And when running 'make test' everything seems to be fine.